File: padroesEstruturais/padrao-de-projeto-composite/php/src/Computador.php
<?php

namespace Composite\Php;

use Composite\Php\Componentes\Componente;
use Ds\Vector;

class Computador
{
    public function __construct(
        private Vector $componentes = new Vector(),
    ) {
    }

    public function setComponentes(Componente $componente): Computador
    {
        $this->componentes->push($componente);
        return $this;
    }

    public function getDescricao(): string
    {
        $saida = "Descrição: \n";
        /** @var Componente $componente */
        foreach($this->componentes as $componente) {
            $saida .= $componente->getDescricao() . "\n";
        }

        return $saida;
    }

    public function getComponentes(): Vector
    {
        return $this->componentes;
    }
}

File: padroesEstruturais/padrao-de-projeto-composite/java/src/main/java/composite/Computador.java
package composite;

import composite.Componentes.Componente;

import java.util.ArrayList;
import java.util.List;

public class Computador {
    private List<Componente> componentes = new ArrayList<Componente>();

    public Computador setComponents(Componente componente) {
        this.componentes.add(componente);
        return this;
    }

    public String getDescricao() {
        StringBuilder saida = new StringBuilder("Descrição: \n");
        for(Componente componentes : this.componentes) {
            saida.append(componentes.getDescricao()).append("\n");
        }

        return saida.toString();
    }

    public List<Componente> getComponents() {
        return this.componentes;
    }
}
